### PR TITLE
[Reland] Avoid saving self for `softmax` and `log_softmax`

### PR DESCRIPTION
**Pull request by soulitzer**
Reland of https://github.com/pytorch/pytorch/pull/65242

The last attempt of the reland automatically rebased onto stable, which did not yet have the revert commit

### CODE REVIEW

**Review by albanD — APPROVED** [review]
stamp

**Comment by JackCaoG** [comment]
Please give me a ping here or at https://github.com/pytorch/xla/pull/3150. I will land xla pr when this one land
